pkg/action/action.go

```diff
@@ -373,6 +373,7 @@ func (cfg *Configuration) recordRelease(r *release.Release) {
 func (cfg *Configuration) Init(getter genericclioptions.RESTClientGetter, namespace, helmDriver string, log DebugLog) error {
 	kc := kube.New(getter)
 	kc.Log = log
+	kc.Namespace = namespace
```

Comment:
Every so often a pull requests comes in looking to set the namespace here. `action.Init` is designed to work with the rest of the _action_ package. `action.Init` can technically be used with multiple Actions without a new instance being created. And, different action runs can have their own namespaces set where Kubernetes resources go. The namespace here is where the Helm secret or configmap is stored instead of where the Kubernetes resources go.

The Install action has an example of how the Namespace is set for Kubernetes resources. For example...

https://github.com/helm/helm/blob/d1595e701f737f1fb367fb6b528c865e43a759ae/pkg/action/install.go#L334

This isn't great for those who want to do something different than what the _action_ package does because of non-exported functions.

We hope to re-design this API at some point. As someone who has built things on top of Helm, I can see where this desire comes from. We are open to bigger changes for Helm v4 and discussing Helm v4.

Until then, this change shouldn't go in because it's going to subtly change behavior for current active users of the Helm SDK.